src/Processors/Formats/Impl/PrettyCompactBlockOutputFormat.cpp

```diff
@@ -69,6 +69,15 @@ void PrettyCompactBlockOutputFormat::writeHeader(
     const Widths & max_widths,
     const Widths & name_widths)
 {
+    if (format_settings.pretty.output_format_pretty_row_numbers)
+    {
+        /// Write left blank
+        for (size_t i = 0; i < row_number_width; ++i)
+        {
+            writeCString(" ", out);
```

Comment:
Why not just `writeString(String(row_numbers_width, ' '), out)` instead of a loop?

Reply:
Also writeChar in a loop will be Ok :)